java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -585,48 +585,48 @@ public ArrowBuf setDouble(int index, double value) {
   @Override
   public ArrowBuf writeShort(int value) {
     ensure(2);
-    PlatformDependent.putShort(addr(writerIndex), (short) value);
-    writerIndex += 2;
+    PlatformDependent.putShort(addr(writerIndex()), (short) value);
+    writerIndex(writerIndex() + 2);
```

Comment:
calling writerIndex() instead of setting the fields adds extra checks. It shows a in your test that we are adding some overhead.
How about you shade by adding a class prefix instead of a package prefix? I think that would remove the need for this.